topology/src/main/java/io/camunda/zeebe/topology/ClusterTopologyManager.java

```diff
@@ -104,7 +98,86 @@ private void setStarted() {
     }
   }
 
-  private void updateLocalTopology(final ClusterTopology topology) throws IOException {
-    persistedClusterTopology.update(topology);
+  ActorFuture<ClusterTopology> onGossipReceived(final ClusterTopology receivedTopology) {
+    final ActorFuture<ClusterTopology> result = executor.createFuture();
+    executor.run(
+        () -> {
+          try {
+            if (receivedTopology != null) {
+              final var mergedTopology =
+                  persistedClusterTopology.getTopology().merge(receivedTopology);
+              if (!mergedTopology.equals(persistedClusterTopology.getTopology())) {
+                LOG.debug(
+                    "Received new topology {}. Updating local topology to {}",
+                    receivedTopology,
+                    mergedTopology);
+                persistedClusterTopology.update(mergedTopology);
+                if (mergedTopology.hasPendingChangesFor(localMemberId)) {
```

Comment:
As discussed, the operation will be only triggered once because the topology is not updated in the meanwhile. But to make it explicit, I now added an in-memory flag to track ongoing operation.